ServerCore/Pages/Puzzles/Checklist.cshtml.cs

```diff
@@ -48,62 +48,48 @@ public async Task<IActionResult> OnGetAsync()
         {
             List<Puzzle> puzzles = await PuzzleHelper.GetPuzzles(_context, Event, LoggedInUser, EventRole);
 
-            Dictionary<int, List<string>> puzzleAuthors = await (from author in _context.PuzzleAuthors
-                                                                        where author.Puzzle.Event == Event
-                                                                        group author by author.PuzzleID into authorList
-                                                                        select new { Puzzle = authorList.Key, Authors = (from a in authorList select a.Author.Name).ToList() }).ToDictionaryAsync(x => x.Puzzle, x => x.Authors);
+            ILookup<int, string> puzzleAuthors = (from author in _context.PuzzleAuthors
+                                                  where author.Puzzle.Event == Event
+                                                  select author).ToLookup(author => author.PuzzleID, author => author.Author.Name);
```

Comment:
Is it a problem that there's no async here?

Reply:
It's probably not a huge deal on the checklist page -- essentially async frees up thread pool threads while the query is running and we're probably not going to run a bunch of checklists in parallel. Doing an async fetch first before converting to other collections is fine though, switching to that.